modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/objectbuilder/ParameterGroupObjectBuilder.java

```diff
@@ -101,22 +84,15 @@ public T build(ResolverSetResult result) throws MuleException {
     }
   }
 
-  private T doBuild(Predicate<String> hasParameter, Function<String, Object> parameters, ValueResolvingContext context,
-                    CursorComponentDecoratorFactory componentDecoratorFactory)
+  private T doBuild(Predicate<String> hasParameter, Function<String, Object> parameters, ValueResolvingContext context)
       throws MuleException {
     T object = createInstance(prototypeClass);
 
     for (FieldElement field : groupDescriptorFields) {
       String name = field.getName();
       if (hasParameter.test(name)) {
-        final boolean isContent = field.isAnnotatedWith(Content.class)
-            || field.isAnnotatedWith(org.mule.sdk.api.annotation.param.Content.class);
-        Object resolvedValue = resolveValue(new StaticValueResolver<>(parameters
-            .apply(name)), context);
-        Object value = context == null || context.resolveCursors()
-            ? resolveCursor(resolvedValue,
-                            isContent ? v -> decorateInput(v, context.getEvent().getCorrelationId(), componentDecoratorFactory)
-                                : identity())
+        Object resolvedValue = resolveValue(new StaticValueResolver<>(parameters.apply(name)), context);
+        Object value = context == null || context.resolveCursors() ? resolveCursor(resolvedValue, MessageUtils::decorateInput)
```

Comment:
Note that now we are mapping with `MessageUtils#decorateInput` every time we do cursor resolution, regardless of the value being content or not. This is because `CursorStream`s need to be wrapped in `UnclosableCursorStream` (see #11209). Please check if this is correct, because maybe I'm doing it in cases in which we shouldn't.

Reply:
same here @ndinu @dabreu

Reply:
same answer as above.

Reply:
same reply

Reply:
same as above.